tenant-config/templates/configmap.yaml

```diff
@@ -0,0 +1,241 @@
+    "$KEYCLOAK_URL/admin/realms/master/users/$ADMIN_USER_ID/groups/$GROUP_ID" \
+    -H "Authorization: Bearer $IDP_AT")
+    
+    if [[ "$RESP" == "204" ]]; then
+      echo "✅ Successfully assigned '$TARGET_GROUP' to '$ADMIN_USER'."
+    else
+      echo "❌ Failed to assign group. HTTP code: $RESP"
+      exit 1
+    fi
+
+    RESP=$(curl -s -o /dev/null -w "%{http_code}" -X PUT ${API_URL}/v1/projects/${DEFAULT_TENANT} \
+    -H "Content-Type: application/json" -H "accept: application/json" \
+    -H "Authorization: Bearer ${IDP_AT}" \
+    -d "{\"description\": \"${DEFAULT_TENANT}\"}")
+    
+    if [[ "$RESP" == "200" ]]; then
+      echo "✅ Successfully created '$DEFAULT_TENANT' project."
+    else
+      echo "❌ Failed to create project. HTTP code: $RESP"
+      exit 1
+    fi
+    
+    PROJ_UUID=$(curl --location ${API_URL}/v1/projects/${DEFAULT_TENANT} \
+    -H "Content-Type: application/json" -H "accept: application/json" \
+    -H "Authorization: Bearer ${IDP_AT}" | jq -r .status.projectStatus.uID)
+
+    while [ "$(curl -s --location ${API_URL}/v1/projects/${PROJ_NAME} \
+      -H "accept: application/json" -H "Content-Type: application/json" \
+      -H "Authorization: Bearer ${IDP_AT}" \
+      | jq -r '.[0].status.projectStatus.statusIndicator')" != "STATUS_INDICATION_IDLE" ]; do
+      echo "Waiting for ${PROJ_NAME} to be provisioned..."
+      sleep 5
+    done
+    
+    echo "✅ Successfully instantiated '$DEFAULT_TENANT' project."
+    
+    EXISTING_USER=$(curl -s -X GET "${KEYCLOAK_URL}/admin/realms/master/users?username=${DEFAULT_USER}" \
+      -H "Authorization: Bearer ${IDP_AT}" \
+      -H "Content-Type: application/json")
+      
+    USER_EXISTS=$(echo "$EXISTING_USER" | jq 'length > 0')
+    
+    if [[ "$USER_EXISTS" == "true" ]]; then
+      echo "ℹ️  User '${DEFAULT_USER}' already exists. Skipping creation."
+      USER_ID=$(echo "$EXISTING_USER" | jq -r '.[0].id')
+    else
+      echo "🔧 Creating user '${DEFAULT_USER}'..."
+      CREATE_RESP=$(curl -s -o /dev/null -w "%{http_code}" -X POST "${KEYCLOAK_URL}/admin/realms/master/users" \
+      -H "Content-Type: application/json" \
+      -H "Authorization: Bearer ${IDP_AT}" \
+      -d "{
+        \"username\": \"${DEFAULT_USER}\",
+        \"email\": \"${DEFAULT_USER}@example.org\",
+        \"emailVerified\": true,
+        \"enabled\": true
+      }")
+  
+      if [[ "$CREATE_RESP" -ne 201 ]]; then
+        echo "❌ Failed to create user. HTTP status: $CREATE_RESP"
+        exit 1
+      fi
+  
+      echo "✅ User '${DEFAULT_USER}' created."
+  
+      # Fetch the new user ID
+      USER_ID=$(curl -s -X GET "${KEYCLOAK_URL}/admin/realms/master/users?username=${DEFAULT_USER}" \
+      -H "Authorization: Bearer ${IDP_AT}" | jq -r '.[0].id')
+  
+      if [[ -z "$USER_ID" || "$USER_ID" == "null" ]]; then
+        echo "❌ Could not fetch user ID after creation."
+        exit 1
+      fi
+    fi
+
+    echo "🔐 Setting password for user ID: $USER_ID..."
```

Comment:
Ensures password is set only if user creation is successful

Reply:
if user is not created the script will exit earlier